contracts/packages/protocol/src/profile.rs

```diff
@@ -0,0 +1,62 @@
+use crate::trade::TradeState;
+use cosmwasm_std::{to_binary, Addr, QuerierWrapper, QueryRequest, StdResult, WasmQuery};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+// Messages
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct InstantiateMsg {}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg {
+    IncreaseTradeCount {
+        profile_address: Addr,
+        final_trade_state: TradeState,
+    },
+    RegisterHub {},
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum QueryMsg {
+    Profile { address: Addr },
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct MigrateMsg {}
+
+// Query Util
+pub fn load_profile(
+    querier: &QuerierWrapper,
+    profile_address: Addr,
+    profile_contract: String,
+) -> Profile {
+    let load_profile_result: StdResult<Profile> =
```

Comment:
this can probably be reduced to fewer lines:
```
querier
        .query(&QueryRequest::Wasm(WasmQuery::Smart {
            contract_addr: profile_contract,
            msg: to_binary(&QueryMsg::Profile {
                address: profile_address.clone(),
            })
            .unwrap(),
        }))
        .unwrap_or(Profile::new(profile_address))
```